Stellarium/stellarium
# Updating StelLogger

### Description
StelLogger previously obtaining many of hardware info through calling CLI tools, and it was caused persistent problem in some environments. Currently, these parts of code rewriting to using platform-specific API/code and format of data has unification form now. Of course, we can expand collected data. 

Information for compiler name and version is coming from CMake now, and all related C++ code was removed.

The code is tested in these operating systems: Windows, macOS, Linux (Ubuntu and Raspberry OS), FreeBSD, OpenBSD, NetBSD, OpenIndiana, Haiku.

### Type of change

- [x] Housekeeping

### How Has This Been Tested?
1. Run Stellarium
2. Open Help windows, Log section 
3. See upper 10-12 lines of log

## Checklist:
